pilot/cmd/pilot-agent/main.go

```diff
@@ -99,12 +99,14 @@ var (
 
 	wg sync.WaitGroup
 
-	instanceIPVar             = env.RegisterStringVar("INSTANCE_IP", "", "")
-	podNameVar                = env.RegisterStringVar("POD_NAME", "", "")
-	podNamespaceVar           = env.RegisterStringVar("POD_NAMESPACE", "", "")
-	istioNamespaceVar         = env.RegisterStringVar("ISTIO_NAMESPACE", "", "")
-	kubeAppProberNameVar      = env.RegisterStringVar(status.KubeAppProberEnvName, "", "")
-	sdsEnabledVar             = env.RegisterBoolVar("SDS_ENABLED", false, "")
+	instanceIPVar        = env.RegisterStringVar("INSTANCE_IP", "", "")
+	podNameVar           = env.RegisterStringVar("POD_NAME", "", "")
+	podNamespaceVar      = env.RegisterStringVar("POD_NAMESPACE", "", "")
+	istioNamespaceVar    = env.RegisterStringVar("ISTIO_NAMESPACE", "", "")
+	kubeAppProberNameVar = env.RegisterStringVar(status.KubeAppProberEnvName, "", "")
+	sdsEnabledVar        = env.RegisterBoolVar("SDS_ENABLED", false, "")
+	autoMTLSEnabled      = env.RegisterBoolVar("ISTIO_AUTO_MTLS_ENABLED", false, "If true, auto mTLS is enabled, "+
```

Comment:
Why just auto mtls and not regular mtls?

Reply:
At least for now, regular mTLS is configured via authn policy, CRD, dynamically. We don't know whether a particular workload needs mTLS or not.

Having that said, we probably can trim lots of logic in future, if we can prove auto mTLS stability and even the authn policy for mTLS can be nuked. That'll be feature improvements.